### PR TITLE
Add campaign based opt outs.

### PR DESCRIPTION
**Pull request by glogiotatidis**
Adds campaign based opt outs, replaces #141

### CODE REVIEW

**Comment by glogiotatidis** [comment]
@ScottDowne please r?

**Comment by ScottDowne** [comment]
Just a heads up I'm still in Hawaii so I'll be doing this review very late
because of time zones.

File: snippets/base/models.py [hunk 1]
```diff
@@ -392,7 +392,8 @@ class Snippet(CachingMixin, SnippetBaseModel):
 
     campaign = models.CharField(
         max_length=255, blank=True, default='',
-        help_text='Optional campaign name. Will be added in the stats ping.')
+        help_text=('Optional campaign name. Will be added in the stats ping. '
+                   'If be used for snippet blocking if set.'))
```
Comment (ScottDowne) [inline snippets/base/models.py]: I think there may be a typo in "If be used for snippet blocking if set."

Not really sure what you were intending though. "To be used for snippet blocking if set."? maybe.

File: snippets/base/templates/base/includes/snippet.js [hunk 2]
```diff
@@ -661,7 +661,9 @@ function popFromBlockList(snippetID) {
 
 function addToBlockList(snippetID) {
     var blockList = getBlockList();
-    snippetID = parseInt(snippetID, 10);
```
Comment (ScottDowne) [inline snippets/base/templates/base/includes/snippet.js]: I cannot come up with a good reason.

Comment (glogiotatidis) [inline snippets/base/templates/base/includes/snippet.js]: `snippetID` used to be a string (now it's an int) and I parsed it as int so it's faster/better to store and lookup.

With this change we still keep snippetID as int and we also allow strings for campaigns.

File: snippets/base/models.py (hunk 1 above)
Comment (glogiotatidis) [inline snippets/base/models.py]: I's going to `will be used...` nice catch, fixed

File: snippets/base/templates/base/includes/snippet.js (hunk 2 above)
Comment (ScottDowne) [inline snippets/base/templates/base/includes/snippet.js]: Makes sense to me.

**Comment by ScottDowne** [comment]
@glogiotatidis this looks good to me. Feel free to land this patch.

Just a heads up, I'll be getting on a plane tonight, so I won't be around to help if anything goes wrong, but I also don't have the ability to back anything out here anyway, so I don't think this changes much.

**Comment by glogiotatidis** [comment]
Cool, I'll merge this but hold pushing until monday so we both have stress free weekend. Safe travels back home and thanks for your time :)

**Pull request merged**